packages/nuxt/src/app/components/nuxt-link.ts

```diff
@@ -280,8 +279,15 @@ export function defineNuxtLink (options: NuxtLinkOptions) {
 
         // Resolves `to` value if it's a route location object
         // converts `""` to `null` to prevent the attribute from being added as empty (`href=""`)
-        const href = typeof to.value === 'object' ? router.resolve(to.value)?.href ?? null : to.value || null
+        let href = typeof to.value === 'object' ? router.resolve(to.value)?.href ?? null : to.value || null
 
+        // joins with `baseURL` if it's an relative link
+        if (href && href.startsWith('/') && !href.startsWith('//')) {
+          const baseURL = useRuntimeConfig().app.baseURL
+          if (href !== '/' && !href.startsWith(baseURL)) {
```

Comment:
What happens if it's a blank target to the homepage there? 🤔

Or I assume we want to check `base` here instead?

Reply:
The '/' doesn't automatically append the baseURL with the current practice, which is indeed a problem.

But then I realized, maybe someone really needs to navigate to a path without baseURL?
Current practice does not seem to allow for this except by using full paths.

Reply:
OK, I think that makes sense, then we just need to update the following for now to my understanding:
```suggestion
          if (baseURL !== '/' && !href.startsWith(baseURL)) {
```